packages/form-plugin/src/actions.ts

```diff
@@ -82,3 +82,11 @@ export class SetFormEnabled {
   }
   constructor(public payload: string) {}
 }
+
+export class ResetForm {
+  static get type() {
+    // NOTE: Not necessary to declare the type in this way in your code. See https://github.com/ngxs/store/pull/644#issuecomment-436003138
+    return '[Forms] Reset Form';
+  }
+  constructor(public payload: { path: string; value?: any; propertyPath?: string }) {}
```

Comment:
The `propertyPath` is unused, please remove it.

Reply:
Thanks for your comment. But this property is used in form.plugin.ts "joinPathWithPropertyPath" function. You can check line 29 in form.plugn.ts. We have a condition:

```typescript
if (type === UpdateFormValue.type || type === UpdateForm.type || type === ResetForm.type) {
      const { value } = event.payload;
      const payloadValue = Array.isArray(value) ? value.slice() : { ...value };
      const path = this.joinPathWithPropertyPath(event);
      nextState = setValue(nextState, path, payloadValue);
    }
```

This action can be used with property value like "UpdateFormValue" and "UpdateForm" actions.

If you think this is not necessary, I can remove it.

Reply:
Yes, please do remove the `propertyPath`. I think that a reset conceptually applies to the whole form therefore giving it a value just for a single internal property does not make sense. Giving it a value to reset the whole form to does make a lot of sense.

Reply:
Yes, please do remove the propertyPath. I think that a reset conceptually applies to the whole form therefore giving it a value just for a single internal property does not make sense. Giving it a value to reset the whole form to does make a lot of sense.